### ISSUE TITLE
Unable to display attestation on mobile - doc inconsistent

### ISSUE THREAD

**Issue by bgrieder** [uport-project/uport-js#49]
We are trying to integrate uPort into Platform 6 for Ondiflo,  a JV between Amalto and Consensys (https://ondiflo.com)

Creating an attestation and displaying it on the mobile app, results in a blank page on the mobile app (android latest play store version - 1.5.297 Alpha).
This happens whether the attestation is pushed as a notification or scanned from a QRCode.

The documentation is inconsistent on attestation creation. It says to use 

     credentials.attest({
       sub: '0x...', // uport address of user
       exp: <future timestamp>, // If your information is not permanent make sure to add an expires timestamp
       claims: {name:'John Smith'}
     }).then(attestation => {
       // send attestation to user
     })

But 
 - both the tests and source show that `claim` should be used instead of `claims`
 - the jsdoc in the code says `claim` should be a `string`, the test and doc says it is an `object`

`* @param    {String}            credential.claim       claim about subject single key value or key mapping to object with multiple values (ie { address: {street: ..., zip: ..., country: ...}})`

Also the following clarifications would be appreciated
 - should the address passed as  `sub` be prefixed with `0x`; In other words is it equal to the `address` received in the profile from the `receive` call  prefixed with `0x` ?
- is the expiry date the number of **seconds** from the epoch ?

Trying various combinations always result in a blank popup on the mobile app.
Could you please clarify the doc and provide a full example with 'real' values.

Many Thanks

**Comment by bgrieder** [comment]
Also, could you please confirm that the Attestation URI to use is:

    me.uport:add?attestation=${attestation}

where `${attestation}` is the value returned by the `attest()` call

**Comment by oed** [comment]
Hey @bgrieder, thanks for spotting the shortcommings in our documentation. 

> * should the address passed as sub be prefixed with 0x; In other words is it equal to the address received in the profile from the receive call prefixed with 0x ?

It should be MNID encoded. But the address that is being encoded should be prefixed with `0x`. For more info check the [uport spec](https://github.com/uport-project/specs/blob/develop/messages/verification.md)
> * is the expiry date the number of seconds from the epoch ?

Yes that's correct!

> Also, could you please confirm that the Attestation URI to use ...

It looks like you are missing an 's' ([verification spec](https://github.com/uport-project/specs/blob/develop/flows/verification.md#send-verifications)). Where did you find the URI without the 's'?

**Comment by bgrieder** [comment]
Hi @oed 

> It should be MNID encoded.

Is the `address` field of the profile returned by the `attest()` call MNID encoded ?


>It looks like you are missing an 's'

This is coming from the doc of the README of uport-js (https://github.com/uport-project/uport-js), paragraph "Attesting information about your users"

`credentials.push(pushToken, `me.uport:add?attestation=${attestationjwt}`, message).then(response => {
})`

**Comment by bgrieder** [comment]
I finally managed to **display** the credential on mobile after scanning a QRCode... but after accepting it, nothing appears in validations not anywhere else.
What am I missing ?

-  I also noticed that the code sample of the doc [there](http://developer.uport.me/guides.html#calling-the-attest-method) calls the method `attestCredentials` when the code shows it is called `attest`

- via push, a message can me passed. What about when scanning a QRCode ?

Is anyone really working on this project ? ( inconsistent docs, very few commits, low activity ... )

**Comment by bgrieder** [comment]
@oed @zachferland please....

**Comment by oed** [comment]
@bgrieder hey, sorry for the slow response.

Seems like the URI is missing a callback url. It needs to be something like:
```js
`me.uport:add?attestations=${encodeURIComponent(jwt)}&callback_url=${encodeURIComponent(topic.url)}`
```

See https://github.com/uport-project/uport-js/issues/44 and https://github.com/uport-project/uport-connect/pull/102

**Comment by bgrieder** [comment]
OK. So I receive a callback with '{'status': 'ok'}' after the user 'attested' on his mobile but then, 
 -  why does this not appear in 'Verfications' in the mobile ?
 -  why do I not get it on as part of the `verified` array when I request a Token again ?

**Comment by oed** [comment]
That you don't get it to show up in 'Verifications' sounds like a bug in the uport app. Can you get verifications from demo.uport.me to show up in your uport app?

**Comment by bgrieder** [comment]
@oed Yes I confirm that I see two (for a certificate and a name) showing when using uPort Demo.
Please note that I am not getting verifications on the server side either on my app.

**Comment by bgrieder** [comment]
@oed Please find a Typescript snippet of the code used to display the attestation

            const attestation = await this.uport.attest(
                {
                    sub:   '0x' + entry.responseToken.address, // uport address of user
                    exp:   new Date().getTime() + 20 * 60 * 1000, // If your information is not permanent make sure to add an expires timestamp
                    claim: { platform6: email }
                }
            )
            const callbackUrl = "http://" + EXTERNAL_HOST + "/_/receiveAttestation?id=" + encodeURIComponent( id )
            const attestationUri = `me.uport:add?attestations=${encodeURIComponent( attestation )}&callback_url=${encodeURIComponent( callbackUrl )}`

            const qrcodeImg = await qrcode.toDataURL( attestationUri, void 0 )

**Comment by oed** [comment]
There are some things you can try:
`Math.floor(new Date().getTime()/1000)` (it might not like decimals)
In:
`sub: '0x' + entry.responseToken.address`
the address should be an MNID so you shouldn't add `0x`. You can verify this by doing `MNID.isMNID(entry.responseToken.address)`

**Comment by bgrieder** [comment]
Thanks the combination of the two fixed it.
